### ISSUE TITLE
Any thoughts of moving Melpa to Codeberg?

### ISSUE THREAD

**Issue by amno1** [melpa/melpa#8216]
Any thoughts of migrating Melpa to Codeberg ?

If I keep my "melpa" repo on GH, but put a project repo itself on the Codeberg, it should still work, right?

Sorry if it is too noob question, but I am not so familiar with hosting and repo administration itself. I do have some thoughts to migrate my stuff to Codeberg.

**Comment by akirak** [comment]
@amno1 Yes, it absolutely works. Even `codeberg` fetcher is now supported in [the recipe format](https://github.com/melpa/melpa#recipe-format). You only need a fork of `melpa` repository on GitHub to send PRs.

**Comment by amno1** [comment]
Thanks for the answer.

**Comment by tarsius** [comment]
Most people seeing the title of this issue would assume that this is about moving away from Github for ideological reasons, but this is not what is actually about. Additionally, the technical misunderstanding has been cleared up. For these reasons I am closing this issue.

**Comment by amno1** [comment]
> Most people seeing the title of this issue would assume that this is about moving away from Github for ideological reasons

Well, that was the exact intention.

I genuinely wonder and asked if you guys have had thoughts of moving Melpa over to Codeberg, exactly for ideological reasons. 

I don't understand why silence, but whatever, I don't think it is neither an unreasonable nor impolite question. If it is, then I apologize, but a simple no, we didn't think about it yet, or we have decided we are happy on GH, or whatever else, would suffice.

The other question was just me, wanted to confirm if things would still work if I move my own stuff on Codeberg, just keeping the Melpa repo on GH, which Akira kindly and promptly answered.

**Comment by tarsius** [comment]
> > Most people seeing the title of this issue would assume that this is about moving away from Github for ideological reasons
> 
> Well, that was the exact intention.
> 
> I genuinely wonder and asked if you guys have had thoughts of moving Melpa over to Codeberg, exactly for ideological reasons.
> 
> I don't understand why silence,

I can only speak for myself. I am trying to take a break and while I am far from actually doing that, I have, for the most part, succeeded in ignoring at least those communications that are neither clear nor actionable.

As I have said, it isn't clear what your motivation is and when you thanked for @akirak's response *without* any indication that it does not actually address your concerns, I took that as an indication that the interpretation them and I independently arrived at was in fact correct.

> The other question was just me,

Ah I see. You asked two questions; one using the title and another using the body. I'm afraid that is not obvious. It is reasonable to ask *additional* questions in the body, but the main question should be at least mentioned in the body and not be confined to the title only. Better yet, it should be expanded on; i.e., say that you are asking the question for ideological reasons and explain why it is a problem that we use Github.

(We have already been accused of "promoting proprietary software" *and* of being "free software zealots", so it is possible that I subconsciously wanted this to be a merely technical question and that that is part of the reason why I interpreted it as such.)

----

> Any thoughts of moving Melpa to Codeberg? [because Github is a proprietary service]

I am neither the person who makes this decision, nor would I be the one who is most affected. That being said:

I am opposed to *removing* Melpa from Github. Most Emacs packages are hosted on Github, so their authors already have an account here, which makes Github the most convenient choice. (You might argue that it would be a good think to force users to open an account on a freedom respecting service, but I fear that would alienate many and thus do more harm than good.)

I have though about suggesting that we *also* add Melpa to Codeberg. Authors who have an account there could then use that instance to open a pull-request with their recipe, while authors that are on Github could continue to use that. But what other forges should be supported, maybe sourcehut? The workflow there is very different and that would be a burden for the person who does most package. So I think @riscy opinion matters the most here; it is him who would have to use different platforms and deal with the differences.

And of course @milkypostman and @purcell might not wish to add Melpa to other forges in the first place.

**Comment by milkypostman** [comment]
I personally have not thought about moving this repository.

I'd be happy to hear reasons why we might want to move, but I don't have the time to do the research in order to build up some motivation. This has worked well for us for many years now and that has a lot of value.

**Comment by amno1** [comment]
> Ah I see. You asked two questions; one using the title and another using the body. I'm afraid that is not obvious. It is reasonable to ask additional questions in the body, but the main question should be at least mentioned in the body and not be confined to the title only. Better yet, it should be expanded on; i.e., say that you are asking the question for ideological reasons and explain why it is a problem that we use Github.

Well, I thought the main question was clear; the other one answered by Akira, is a related one, and partly me not being sure for 100%, partly also adding in to the main question that it is possible to have GH and CB co-existing with each other. 

> the main question should be at least mentioned in the body and not be confined to the title only

Sure, I understand, and can agree with that. I am sorry, if it appears not so obvious; people perceive what is obvious/non-obvious differently, so it is me being clumsy. Not my intention, of course, but so it happened this time, seems like. I was like more genuinely just interested to hear what is Melpa devs/community opinions on the issue. Could have made it more clear in the message; I was just fast, didn't think it would be a big deal. I also understand the issue is somewhat uncomfortable.

> explain why it is a problem that we use Github

I am not convinced myself if it is a problem or not, to be honest. I am trying to make some definitive opinion, but I am not there yet. Partly a reason why I ask.

> We have already been accused of "promoting proprietary software" and of being "free software zealots", so it is possible that I subconsciously wanted this to be a merely technical question and that that is part of the reason why I interpreted it as such.

Yeah, I know; I just read through that, I don't care about such things. I myself was accused of using non-free OS (Arch Linux instead of some *gnu free* that J-L use) in a conversation on emacs-devel once :).  

For me personally, I understand that we (users of GH) are the product, the moment GH started out, long before MS bought it. Of course, they have to make money somehow, they are a company, and companies are in the business of making money. So it is OK. They provide free services, so it was a bit of symbiosis. However, the issue has become quite different with their AI. I don't think that my Emacs code is worth much to anyone, it is just toys I do as a hobby, but it is the principle. One can make money in different ways. Just blatantly taking people's code without respecting authors and copyrights seems a bit over the edge. At the same time I am aware they can just harvest Codeberg, Gitlab and any other public repo, so it is a bit *wtf* situation. 

I understand that dealing with different platforms would be extra work for mods, having two Melpas (Melpa in two different places) is probably not very desirable for the maintenance. 

Yes, Sourcehut is so very different; I feel more at home with Codeberg since I am familiar with GH already, I don't know if so is for other users.

> I'd be happy to hear reasons why we might want to move

I am not sure if there are practical reasons, mostly political. People seem to be pissed off with Copilot and are migrating to other platforms or to private hosting. GH still has the momentum in number of users, and it seems it is still "the standard" platform to share the code, at least what I see when reading about new projects in various Reddit forums I visit.

**Comment by riscy** [comment]
Apologies for not seeing to this one sooner --

> So I think @riscy opinion matters the most here; it is him who would have to use different platforms and deal with the differences.

> I understand that dealing with different platforms would be extra work for mods, having two Melpas (Melpa in two different places) is probably not very desirable for the maintenance.

When I'm reviewing, I'm really just looking at a recipe and a dialogue with the maintainer(s) -- and I don't raise issues or PRs to the upstreams.  I have about a dozen "saved replies" on GitHub and use that feature about once every couple of weeks.  So it's fairly simple and there isn't much technical challenge in moving between SCMs or even turning the process into a mailing list.  But it's still simpler to have a single platform.

**Comment by Fuco1** [comment]
BTW even RMS himself on one of the Emacsconfs (2020?) said that GitHub is not an issue.  It's just a git repo.  It forces some JS on you but you don't have to use GitHub, you can clone the repo and PR from magit :)
